src/pushprocessor.ts

```diff
@@ -433,8 +433,74 @@ export class PushProcessor {
         return PushProcessor.cachedGlobToRegex[glob];
     }
 
+    /**
+     * Parse a dotted key into parts following the escape rules.
+     *
+     * @param str - The key of the push rule condition: a dotted field.
+     * @returns The unescaped parts to fetch.
+     */
+    private partsFotDottedKey(str: string): string[] {
+        const result = [];
+
+        // The current field and whether the previous character was the escape
+        // character (a backslash).
+        let part = "";
+        let escaped = false;
+
+        // Iterate over each character, and decide whether to append to the current
+        // part (following the escape rules) or to start a new part (based on the
+        // field separator).
+        for (let i = 0; i < str.length; ++i) {
+            const c = str[i];
+
+            // If the previous character was the escape character (a backslash)
+            // then decide what to append to the current part.
+            if (escaped) {
+                if (c === "\\" || c === ".") {
+                    // An escaped backslash or dot just gets added.
+                    part += c;
+                } else {
+                    // A character that shouldn't be escaped gets the backslash prepended.
+                    part += "\\" + c;
+                }
+                // This always resets being escaped.
+                escaped = false;
+                continue;
+            }
+
+            if (c == ".") {
+                // The field separator creates a new part.
+                result.push(part);
+                part = "";
+            } else if (c == "\\") {
+                // A backslash adds no characters, but starts an escape sequence.
+                escaped = true;
+            } else {
+                // Otherwise, just add the current character.
+                part += c;
+            }
+        }
+
+        // Ensure the final part is included, if there is one.
+        if (part.length) {
```

Comment:
Technically if part is empty, pushing it doesn't do any harm, right? So I might be tempted to remove the if (but just a thought).

Reply:
Good spot -- this was a bug. There's no reason you can't have empty parts here, e.g. `m..foo` to match:

```json
{
  "m": {
    "": {
      "foo"
    }
  }
}
```

Definitely a (silly) edgecase, but it simplifies the code to support it!

Reply:
Brilliant! I didn't spot that, but am happy I perturbed your brain sufficiently for you to :-)